compiler/circle-interpreter/src/CircleInterpreter_cffi.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This library is for using luci-interpreter in Python. It is implemented
+ *  with CFFI. CFFI is a FFI (Foreign Function Interface) for Python calling
+ *  C code.
+ */
+
+#include <cstddef>
+#include <string>
+
+#include <luci/Importer.h>
+#include <luci_interpreter/Interpreter.h>
+
+namespace
+{
+
+// Global variable for error message
+std::string last_error_message;
+
+template <typename NodeT> size_t getTensorSize(const NodeT *node)
+{
+  uint32_t tensor_size = luci::size(node->dtype());
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    tensor_size *= node->dim(i).value();
+  return tensor_size;
+}
+
+// Function to retrieve the last error message.
+extern "C" const char *get_last_error(void) { return last_error_message.c_str(); }
+
+// Clear the last error message
+extern "C" void clear_last_error(void) { last_error_message.clear(); }
```

Comment:
What are these functions for? Is it used in Python-side? If so, could you show the Python codes too?

Reply:
Here's the python codes.
```python
def check_for_errors():
  error_message = ffi.string(C.get_last_error()).decode('utf-8')
  if error_message:
    C.clear_last_error()
    raise RuntimeError(f'C++ Exception: {error_message}')

# ..

try:
  C.Interpreter_writeInputTensor(intp, 0, c_arr1, 123456)
  check_for_errors()
except RuntimeError as e:
  print(e)
```